src/Devtech/Common/Validator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: zoran.stankovic
+ * Date: 1/29/2016
+ * Time: 1:20 PM
+ */
+
+namespace Devtech\Common;
+
+
+class Validator implements IValidator
```

Comment:
I would recommend creating validator class for each of the types which can be configurable.
eg. You can have IntegerValidator class that validated whether the value is integer and can be configurable - like max/min length/value. This validator can be then passed as parameter for each parameter validation.